docs/minio-job.md

```diff
@@ -0,0 +1,208 @@
+      args:
+        user: ${USER}
+        password: ${PASSWORD}
+      envFrom:
+        - secretRef:
+            name: mytestsecretenvs
+    - name: add-my-user-2
+      op: admin/user/add
+      args:
+        user: pedro
+        password: $PASSWORD
+      env:
+        - name: PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: mytestsecret
+              key: PASSWORD
+    - name: add-my-policy
+      op: admin/policy/create
+      args:
+        name: memes-access
+        policy: /temp/policy.json
+      volumeMounts:
+        - name: policy
+          mountPath: /temp
+      volumes:
+        - name: policy
+          configMap:
+            name: mytestconfig
+            items:
+              - key: policy.json
+                path: policy.json
+    - op: admin/policy/attach
+      dependsOn:
+        - add-my-user-1
+        - add-my-user-2
+        - add-my-policy
+      args:
+        policy: memes-access
+        user: daniel
+    - op: admin/policy/attach
+      dependsOn:
+        - add-my-user-1
+        - add-my-user-2
+        - add-my-policy
+      args:
+        policy: memes-access
+        user: pedro
+    - op: stat
+      command:
+        - "mc"
+        - "stat"
+        - "myminio/memes"
+```
+The MinioJob is a Kubernetes Job that runs mc commands. It uses the MinIO client (mc) to interact with the MinIO server.
+## mcImage
+The `mcImage` field specifies the Docker image that will be used to run the mc commands. In this case, the image is `minio/mc:latest`.
+## serviceAccountName
+The `serviceAccountName` field specifies the name of the Kubernetes ServiceAccount that will be used to run the mc commands. In this case, the ServiceAccount is `mc-job-sa`.
+## securityContext
+example:
+```yaml
+runAsUser: 1000
+runAsGroup: 1000
+fsGroup: 1000
+fsGroupChangePolicy: "OnRootMismatch"
+runAsNonRoot: true
+allowPrivilegeEscalation: false
+capabilities:
+  drop:
+    - ALL
+```
+The `securityContext` field specifies the security context that will be used to run the mc commands. 
+## containerSecurityContext
+The `containerSecurityContext` field specifies the security context that will be used to run the `mc` commands in the container.
+## tenant
+```yaml
+name: tenantName
+namespace: tenantNamespace
+```
+The target tenant that the job will run against.
+## commands
+### args
+if you set this field, the `mc` command will be executed with the arguments.
+`op` must be one of these:
+`make-bucket`, `admin/user/add`,`admin/policy/create`,`admin/policy/attach`, `admin/config/set`, `support/callhome`,`license/register`
+```yaml
+op: make-bucket
+args:
+  name: memes
+```
+Will do a job like `mc mb myminio/memes`
+```yaml
+name: add-my-policy
+op: admin/policy/create
+args:
+  name: memes-access
+policy: /temp/policy.json
+volumeMounts:
+- name: policy
+  mountPath: /temp
+volumes:
+- name: policy
+  configMap:
+    name: mytestconfig
+    items:
+      - key: policy.json
+        path: policy.json
```

Comment:
Assuming the complexity here is because of how Kubernetes expects us to access/mount the ConfigMap.

I'm guessing we cannot use the same 'secrets' pathway here where we just pass the entire JSON body as a variable? Mostly asking from a 'can we make this simpler' perspective - for the user it might not matter much that the YAML here is a little heavy handed.

Reply:
Although it would be technically possible to make this simpler, we choose to use this approach that:
- Looks familiar to people who are familiar with Kubernetes. If you would like to create a pod manually and mount a policy in there, you would do it like this.
- Keep the flexibility that Kubernetes provide. We had a simpler syntax in the beginning, but it was quite limited.